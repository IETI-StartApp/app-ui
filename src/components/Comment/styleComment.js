import styled from 'styled-components';

export const ButtonDesign = styled.button`
width: 50%;
height: 50px;
margin-top: 10px;
border-radius: 2px;
background: linear-gradient(90deg,
rgb(149, 7, 64) 0%,
rgb(127, 6, 54) 100%);
outline: none;
border: none;
align-content: center;
color: #fff;
font-size: 1rem;

`;
export const FormDesign = styled.form`
`;