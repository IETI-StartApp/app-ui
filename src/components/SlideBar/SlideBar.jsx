
import React from "react";
import Carousel from "react-elastic-carousel";

const items = [
    { id: 1, alt: "", src: "https://media.discordapp.net/attachments/781412743537491969/830539929565528115/unknown.png?width=250&height=250" },
    { id: 2, alt: "", src: "https://media.discordapp.net/attachments/781412743537491969/830540011442405457/unknown.png?width=250&height=250" },
    { id: 3, alt: "", src: "https://media.discordapp.net/attachments/781412743537491969/830541738476306432/unknown.png?width=250&height=250" },
    { id: 4, alt: "", src: "https://media.discordapp.net/attachments/781412743537491969/830541868105203732/unknown.png?width=250&height=250" },
    { id: 5, alt: "", src: "https://media.discordapp.net/attachments/781412743537491969/830542506985914378/unknown.png?width=250&height=250" },
    { id: 6, alt: "", src: "https://media.discordapp.net/attachments/781412743537491969/830542595431727134/unknown.png?width=250&height=250" },
    { id: 7, alt: "", src: "https://media.discordapp.net/attachments/781412743537491969/830543249060003840/unknown.png?width=250&height=250" }
]
const slide = items.map((item) =>
    <div key={item.id}><img src={item.src} alt={item.alt} /></div>
);


export const SlideBar = () => {
    return (
        <div className="App">
            <div className="controls-wrapper">
            </div>
            <hr className="seperator" />
            <div className="carousel-wrapper">
                <Carousel>
                    {slide}
                </Carousel>

            </div>
        </div>
    );
}

