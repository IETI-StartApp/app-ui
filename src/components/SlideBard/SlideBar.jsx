import React from 'react'

const SlideBar = () => {
    return (
        <div>
            Slide bar aca
        </div>
    )
}

export default SlideBar
