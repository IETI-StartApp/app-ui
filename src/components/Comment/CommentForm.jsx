import React, { useState } from 'react'
import Rating from '@material-ui/lab/Rating';
import { Grid, TextField } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import "./styleComment.js";
import { ButtonDesign, FormDesign } from "./styleComment";



const useStyles = makeStyles((theme) => ({
    root: {
        display: 'flex',
        flexDirection: 'column',
        width: 200,
        '& > * + *': {
            marginTop: theme.spacing(1),
        },
    },
}));

export const Comment = () => {

    const classes = useStyles();
    const [values, setValues] = useState({
        showComments: false,
        id: 0,
        author: "",
        data: ""
    })
    const [qualifier, setQualifier] = useState(null)
    const handleChange = e => {
        setValues({
            ...values, [e.target.name]: e.target.value
        })
    }
    const addComment = (author, body, totalRate) => {
        const rating = {
            "author": author,
            "data": body,
            "rate": totalRate
        }
        alert(author + " Acabamos de registrar tu comentario y gracias por la calificacion de " + qualifier)
        
    }
    
    const handleSubmit = (event) => {
        event.preventDefault();
        addComment(values.author, values.data, qualifier);
    }
    const ratingChanged = (e, newRating) => {
        setQualifier(newRating);
        
    };
    return (
        <div className="comment-box">
            <Grid container
                direction="row"
                justify="flex-end"
                alignItems="stretch"
                style={{ overflowY: "auto" }}>
                <Grid item xs={8}>
                    <h1>Califica el proyecto y deja un comentario </h1>
                    <FormDesign className="form" onSubmit={handleSubmit}>
                        <Grid item xs={11}>
                            <div className={classes.root}>
                                <Rating align-content="center" size="large" name="qualifier" precision={1} value={qualifier} onChange={ratingChanged} />
                            </div>
                            <div className="comment-form-fields">
                                <TextField style={{
                                    width: "50%",
                                    margin: "1rem 0",
                                }} placeholder="Name" required name="author" value={values.author} onChange={handleChange} /><br />

                                <TextField style={{
                                    width: "50%",
                                    margin: "1rem 0",
                                }}
                                    id="standard-multiline-static"
                                    label="Comment"
                                    multiline
                                    rows={4}
                                    defaultValue="Default Value"
                                    required name="data" value={values.data} onChange={handleChange}
                                />                                
                            </div>
                            <div className="comment-form-actions">
                                <ButtonDesign className="button" type="submit">
                                    Â¡Comentar!
                        </ButtonDesign>
                            </div>
                            <h2>Tus comentarios ayudaran a la mejora del proyecto!</h2>
                        </Grid>
                    </FormDesign>
                </Grid>
            </Grid>
        </div>
    )
}

