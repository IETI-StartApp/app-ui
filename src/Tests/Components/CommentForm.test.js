import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { render, screen } from '@testing-library/react';
import { Comment } from '../../components/Comment/CommentForm';
import { SlideBar } from '../../components/SlideBar/SlideBar';

describe('Testing comment page', () => {
    const wrapper = shallow(<Comment />);
    const wrapper2 = shallow(<SlideBar />);
    test('should render comment page', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('should render comment page', () => {
        expect(wrapper2).toMatchSnapshot();
    });

    test('deberia mostrar el comentar', () => {
        const btn = wrapper.find('ButtonDesign')
    });
    test('deberia mostrar el comentar', () => {
        const mockedEvent = { target: {}, preventDefault: () => { } }
        const spy = jest.spyOn(wrapper.instance(), 'onSubmit')
        wrapper.find('form').simulate('submit', mockedEvent)
        expect(spy).toHaveBeenCalled()
    });
    test('renders learn react link', () => {
        render(<Comment />);
        const linkElement = screen.getByText(/Califica el proyecto y deja un comentario/i);
        expect(linkElement).toBeInTheDocument();
    });

    test('renders learn react link', () => {
        render(<Comment />);
        const linkElement = screen.getByText(/Tus comentarios ayudaran a la mejora del proyecto!/i);
        expect(linkElement).toBeInTheDocument();
    });
    test('renders learn react link', () => {
        render(<Comment />);
        const linkElement = screen.getByText(/Â¡Comentar!/i);
        expect(linkElement).toBeInTheDocument();
    });
})