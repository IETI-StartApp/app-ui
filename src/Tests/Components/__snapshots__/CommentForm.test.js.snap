// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing comment page should render comment page 1`] = `
<div
  className="comment-box"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="stretch"
    container={true}
    direction="row"
    justify="flex-end"
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={8}
    >
      <h1>
        Califica el proyecto y deja un comentario 
      </h1>
      <styled.form
        className="form"
        onSubmit={[Function]}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={11}
        >
          <div
            className="makeStyles-root-1"
          >
            <WithStyles(ForwardRef(Rating))
              align-content="center"
              name="qualifier"
              onChange={[Function]}
              precision={1}
              size="large"
              value={null}
            />
          </div>
          <div
            className="comment-form-fields"
          >
            <WithStyles(ForwardRef(TextField))
              name="author"
              onChange={[Function]}
              placeholder="Name"
              required={true}
              style={
                Object {
                  "margin": "1rem 0",
                  "width": "50%",
                }
              }
              value=""
            />
            <br />
            <WithStyles(ForwardRef(TextField))
              defaultValue="Default Value"
              id="standard-multiline-static"
              label="Comment"
              multiline={true}
              name="data"
              onChange={[Function]}
              required={true}
              rows={4}
              style={
                Object {
                  "margin": "1rem 0",
                  "width": "50%",
                }
              }
              value=""
            />
          </div>
          <div
            className="comment-form-actions"
          >
            <styled.button
              className="button"
              type="submit"
            >
              ¡Comentar!
            </styled.button>
          </div>
          <h2>
            Tus comentarios ayudaran a la mejora del proyecto!
          </h2>
        </WithStyles(ForwardRef(Grid))>
      </styled.form>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`Testing comment page should render comment page 2`] = `
<div>
  Imagenes
</div>
`;

exports[`Testing register page should render register page 1`] = `
<div
  className="comment-box"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="stretch"
    container={true}
    direction="row"
    justify="flex-end"
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={8}
    >
      <h1>
        Califica el proyecto y deja un comentario 
      </h1>
      <styled.form
        className="form"
        onSubmit={[Function]}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={11}
        >
          <div
            className="makeStyles-root-1"
          >
            <WithStyles(ForwardRef(Rating))
              align-content="center"
              name="qualifier"
              onChange={[Function]}
              precision={1}
              size="large"
              value={null}
            />
          </div>
          <div
            className="comment-form-fields"
          >
            <WithStyles(ForwardRef(TextField))
              name="author"
              onChange={[Function]}
              placeholder="Name"
              required={true}
              style={
                Object {
                  "margin": "1rem 0",
                  "width": "50%",
                }
              }
              value=""
            />
            <br />
            <WithStyles(ForwardRef(TextField))
              defaultValue="Default Value"
              id="standard-multiline-static"
              label="Comment"
              multiline={true}
              name="data"
              onChange={[Function]}
              required={true}
              rows={4}
              style={
                Object {
                  "margin": "1rem 0",
                  "width": "50%",
                }
              }
              value=""
            />
          </div>
          <div
            className="comment-form-actions"
          >
            <styled.button
              className="button"
              type="submit"
            >
              ¡Comentar!
            </styled.button>
          </div>
          <h2>
            Tus comentarios ayudaran a la mejora del proyecto!
          </h2>
        </WithStyles(ForwardRef(Grid))>
      </styled.form>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
